// react-landing-page/dist/atoms/Section.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _rebass = require('rebass');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Section = function Section(_ref) {
  var heading = _ref.heading,
      subhead = _ref.subhead,
      subheadOpacity = _ref.subheadOpacity,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ['heading', 'subhead', 'subheadOpacity', 'children']);

  return _react2.default.createElement(
    _rebass.Flex,
    _extends({
      py: [4, 5],
      px: [1, 1, 2, 2, 3],
      flexDirection: 'column',
      alignItems: 'center'
    }, props),
    _react2.default.createElement(
      _rebass.Heading,
      { textAlign: 'center', mb: [1] },
      heading
    ),
    _react2.default.createElement(
      _rebass.Subhead,
      { textAlign: 'center', mb: [1, 2, 3, 4, 5],
        style: {
          opacity: subheadOpacity
        } },
      subhead
    ),
    children
  );
};

Section.propTypes = {
  heading: _propTypes2.default.string,
  subhead: _propTypes2.default.string,
  subheadOpacity: _propTypes2.default.number,
  children: _propTypes2.default.any
};
Section.defaultProps = {
  subheadOpacity: 0.5
};

exports.default = Section;
